### PR TITLE
Upgrade Calico to 3.7.4

### PR DESCRIPTION
**Pull request by lzang**
What type of PR is this?
/kind feature

What this PR does / why we need it:
Upgrade calico network policy controller to 3.7.4

Does this PR introduce a user-facing change?:
NONE

### CODE REVIEW

**Comment by bowei** [comment]
Add link to calico release notes for this version in the commit comment.

**Comment by lzang** [comment]
When I tested this on master branch, calico node comes up fine but vertical autoscaler is crashlooping. 
$ kubectl get pod --all-namespaces |grep calico
kube-system   calico-node-n9fx8                                     1/1     Running            0          16h
kube-system   calico-node-v5j5g                                     1/1     Running            0          16h
kube-system   calico-node-vertical-autoscaler-6649bb556f-b6nvx      0/1     CrashLoopBackOff   200        16h
kube-system   calico-typha-7574c556d8-tkjrn                         1/1     Running            0          16h
kube-system   calico-typha-horizontal-autoscaler-86f6f9d548-jfgc5   1/1     Running            0          16h
kube-system   calico-typha-vertical-autoscaler-8679cfbd8d-8wkb5     0/1     CrashLoopBackOff   200        16h

The log indicates that the failure reason is because now the deployment changes to apps/v1, and seems not supported by the current autoscaler.

E0729 17:38:26.932770       1 autoscaler.go:49] no supported API group for target: map[apps/v1:true]

Since the issue is not caused by current change, will file a separate bug to track it.

**Review by caseydavenport — COMMENTED** [review]
> no supported API group for target: map[apps/v1:true]

Ahh yeah, that's too bad. We should definitely update the autoscalers to support the v1 APIs here since the old ones are on the way out.

In the meantime, we can probably keep using the extensions API if we're blocked here. It's not required on the Calico side, but trying to stay ahead of the pending deprecation. 

Overall this PR looks good to me - I had just one minor cosmetic comment. I'll run this myself as well once any final changes regarding the autoscalers are added. 

@lzang thanks for putting this together

File: cluster/addons/calico-policy-controller/calico-node-daemonset.yaml [hunk 1]
```diff
@@ -96,52 +146,11 @@ spec:
             - mountPath: /var/lib/calico
               name: var-lib-calico
               readOnly: false
+            - mountPath: /run/xtables.lock
+              name: xtables-lock
+              readOnly: false
         # This container installs the Calico CNI binaries
         # and CNI network config file on each node.
```
[the remark this answers is not in the document]
Comment (lzang) [reply cluster/addons/calico-policy-controller/calico-node-daemonset.yaml]: Thanks! Updated the patch.

**Comment by bowei** [comment]
Thanks for reviewing @caseydavenport

**Pull request merged**